platform/pulp/platform/models/task.py

```diff
@@ -0,0 +1,296 @@
+    :type task_type: models.TextField
+
+    :cvar task_group: The group this task belongs to
+    :type task_group: models.UUIDField
+
+    :cvar state: The state of the task
+    :type state: models.TextField
+
+    :cvar start_time: The time the task started executing
+    :type start_time: models.DateTimeField
+
+    :cvar finish_time: The time the task finished executing
+    :type finish_time: models.DateTimeField
+
+    :cvar error: Serialized collection of errors that might have occurred while task was running
+    :type error: models.TextField
+
+    :cvar result: Serialized return value of the task
+    :type result: models.TextField
+
+    Relations:
+
+    :cvar parent: Task that spawned this task (if any)
+    :type parent: models.ForeignKey
+
+    :cvar worker: The worker that this task is in
+    :type worker: models.ForeignKey
+    """
+
+    WAITING = 'waiting'
+    SKIPPED = 'skipped'
+    ACCEPTED = 'accepted'
+    RUNNING = 'running'
+    SUSPENDED = 'suspended'
+    FINISHED = 'finished'
+    ERROR = 'error'
+    CANCELED = 'canceled'
+    STATES = (
+        (WAITING, 'Waiting'),
+        (SKIPPED, 'Skipped'),
+        (ACCEPTED, 'Accepted'),
+        (RUNNING, 'Running'),
+        (SUSPENDED, 'Suspended'),
+        (FINISHED, 'Finished'),
+        (ERROR, 'Error'),
+        (CANCELED, 'Canceled')
+    )
+    task_type = models.TextField()
+    task_group = models.UUIDField()
+    state = models.TextField(choices=STATES)
+
+    start_time = models.DateTimeField()
+    finish_time = models.DateTimeField()
+
+    _error = models.TextField(null=True, db_column='error')
+    _result = models.TextField(null=True, db_column='result')
+
+    parent = models.ForeignKey("Task", null=True)
+    worker = models.ForeignKey(Worker)
+
+    @property
+    def result(self):
+        try:
+            return self._result_object
+        except AttributeError:
+            self._result_object = json.loads(self._result)
+            return self._result_object
+
+    @property
+    def error(self):
+        try:
+            return self._error_object
+        except AttributeError:
+            self._error_object = json.loads(self._result)
+            return self._error_object
+
+
+    def save(self, *args, **kwargs):
+        try:
+            self._args = json.dumps(self._error_object)
+        except AttributeError:
+            pass
+
+        try:
+            self._kwargs = json.dumps(self._result_object)
+        except AttributeError:
+            pass
+
+        super(Model, self).save(*args, **kwargs)
+
+
+class TaskTags(Model):
+    """
+    Custom tags for a task
+
+    Fields:
+
+    :cvar name: The name of the tag
+    :type name: models.TextField
+
+    Relations:
+
+    :cvar task: The task this tag is associated with
+    :type task: models.ForeignKey
+    """
+    name = models.TextField()
+
+    task = models.ForeignKey(Task, related_name="tags", related_query_name="tag")
+
+class ScheduledCalls(Model):
```

Comment:
This model was never converted to mongoengine, and there were statements that the pymongo object had values that no longer were used.  This model should probably be scrutinized heavily to ensure data wasn't missed.  I did a best guess between what's in the pymongo object and what is stored in the database when a schedule is created.